test/xpu_api/numerics/complex.number/cases.h

```diff
@@ -218,48 +215,51 @@ CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
         return non_zero_nan;
     }
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
 inline
 int
 classify(double x)
 {
-// Suppress clang warning: comparison with infinity always evaluates to false in fast floating point modes [-Wtautological-constant-compare]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
-
     if (x == 0)
         return zero;
     if (std::isinf(x))
         return inf;
     if (std::isnan(x))
         return NaN;
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
-void is_about(float x, float y)
+auto
+get_eps(float)
 {
-// Suppress clang warning: floating-point comparison is always true; constant cannot be represented exactly in type 'float' [-Wliteral-range]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_LITERAL_RANGE
-
-    assert(std::abs((x-y)/(x+y)) < 1.e-6);
+    constexpr float eps = 1.e-6;
+    return eps;
+};
 
-CLANG_DIAGNOSTIC_POP
-}
+template <typename T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, T>::type
+get_eps(T)
+{
+    static_assert(::std::is_same_v<T, double> || ::std::is_same_v<T, long double>);
+    constexpr T eps = 1.e-11;
+    return eps;
+};
 
-void is_about(double x, double y)
+template <typename T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, void>::type
+is_about(T x, T y)
 {
-    assert(std::abs((x-y)/(x+y)) < 1.e-14);
+    const auto exp = get_eps(T{});
+    assert(std::abs(x - y) < exp);
```

Comment:
std::abs for floating point numbers is supported just from C++23..
https://en.cppreference.com/w/cpp/numeric/math/fabs

Reply:
They are simple ```constexpr``` since C++23